vhdbuilder/packer/cleanup.sh

```diff
@@ -181,7 +181,58 @@ if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN,,}" == "false" ]]; then
   set -x
 fi
 
-#clean up storage account created over a week ago
+# attempt to clean up Windows managed images and SIG image versions created over a week ago in SIG_GALLERY_NAME (except the production gallery)
+if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN}" == "False" && "${SIG_GALLERY_NAME}" != "AKSWindows" ]]; then
+  echo "Looking for Windows managed images in ${AZURE_RESOURCE_GROUP_NAME} created over ${EXPIRATION_IN_HOURS} hours ago..."
+
+  managed_image_ids=""
+  sig_version_ids=""
+
+  # delete outdated Windows managed images and associated SIG versions (.tags.os must be "Windows")
+  images=$(az image list -g ${AZURE_RESOURCE_GROUP_NAME} | jq --arg dl $deadline '.[] | select(.tags.os == "Windows") | select(.tags.now < $dl).name' | tr -d '\"' || "")
+  for image in $images; do
```

Comment:
Do we only delete the managed images and sig versions for the last image since line 199-211 is not in the for loop?

Reply:
199-211 use the string of managed image and SIG version IDs to actually run the delete operation, the loop just builds up those strings